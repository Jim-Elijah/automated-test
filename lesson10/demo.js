import axios from 'axios'

export const fetchData = () => {
  return axios.get('/').then(res => res.data)
}
// 假设 返回下面内容
// `(function(){return '123'})()`

export const getNumber = () => {
  return 123
}