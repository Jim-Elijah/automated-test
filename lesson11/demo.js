// export const timers = (cb) => {
//   setTimeout(() => {
//     cb()
//   }, 3000);
// }

export const timers = (cb) => {
  setTimeout(() => {
    cb()
    setTimeout(() => {
      cb()
    }, 3000);
  }, 3000);
}