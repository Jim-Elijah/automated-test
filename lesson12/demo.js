import Util from './util.js'

const demoFunction = () => {
  const util = new Util()
  util.a()
  util.b()
}

export default demoFunction