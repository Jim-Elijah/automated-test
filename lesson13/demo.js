import $ from 'jquery'

const append2Body = () => {
  $('body').append('<div></div>')
}

export default append2Body