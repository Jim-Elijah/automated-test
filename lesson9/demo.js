export const generateConfig = () => {
  return {
    server: 'http://locaohost',
    port: 8080,
    domain: 'localhost',
    time: new Date(),
  }
}

export const generateAnotherConfig = () => {
  return {
    server: 'http://locaohost',
    port: 8081,
    domain: 'localhost',
    time: new Date(),
  }
}