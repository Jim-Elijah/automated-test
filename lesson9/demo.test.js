import { generateConfig, generateAnotherConfig } from "./demo";

test("测试 generateConfig", () => {
  expect(generateConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    {
      "domain": "localhost",
      "port": 8080,
      "server": "http://locaohost",
      "time": Any<Date>,
    }
  `
  );
});

test("测试 generateAnotherConfig", () => {
  expect(generateAnotherConfig()).toMatchSnapshot({
    time: expect.any(Date),
  });
});
