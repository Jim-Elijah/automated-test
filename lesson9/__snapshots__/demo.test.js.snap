// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateAnotherConfig 1`] = `
{
  "domain": "localhost",
  "port": 8081,
  "server": "http://locaohost",
  "time": Any<Date>,
}
`;
